File: java/liskov-start/src/test/java/no/bouvet/solid/lsp/BirdServiceTest.java
package no.bouvet.solid.lsp;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;

public class BirdServiceTest {
    private BirdService subject;

    @Before
    public void setUp() {
        subject = new BirdService();
    }

    @Test
    public void testFeedThemAll() {
        subject.feedThemAll();
    }

    @Test
    public void testLetThemGo() {
        subject.letThemGo();
    }

}

File: dotnet/Solid/Tests/LSP/Example1/BirdServiceTest.cs
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Solid.LSP.Example1;

namespace Tests.LSP.Example1
{
    [TestClass]
    public class BirdServiceTest
    {
        private BirdService _subject;


        [TestInitialize]
        public void Setup()
        {
            _subject = new BirdService();
        }

        [TestMethod]
        public void TestFeedThemAll()
        {
            _subject.FeedThemAll(AllBirds());
        }

        [TestMethod]
        public void TestLetThemGo()
        {
            _subject.LetThemGo(AllBirds());
        }

        private List<Bird> AllBirds()
        {
            return new List<Bird>
                   {
                        new Duck(),
                        new Penguin(),
                        new Crow()
                   };
        }
    }
}
